src/core/device.c

```diff
@@ -90,6 +92,46 @@ static int device_set_sysfs(Device *d, const char *sysfs) {
         return 0;
 }
 
+static void io_cost_qos_reset(IOCostQos *io_cost_qos) {
+        assert(io_cost_qos);
+
+        io_cost_qos->enabled = -1;
+        io_cost_qos->ctrl = _IO_COST_CTRL_INVALID;
+        io_cost_qos->read_latency_percentile = 0;
+        io_cost_qos->read_latency_threshold = 0;
+        io_cost_qos->write_latency_percentile= 0;
+        io_cost_qos->write_latency_threshold= 0;
+        io_cost_qos->min= 0;
+        io_cost_qos->max= 0;
```

Comment:
These days we prefer using compound literals to initialize structs. Can we rewrite this as:

```
*io_cost_qos = (IOCostQos) {
        .enabled = -1,
        .ctrl = _IO_COST_CTRL_INVALID
};
```

Compound literals will assign zero to any fields that aren't explicitly initialized.

Reply:
This code is not meant to be initializing the struct. The struct itself must already have been initialized (hence the `assert(io_cost_qos)` above). If we were to use a compound literal here wouldn't we first have to free the existing object?

Reply:
We wouldn't have to free/malloc anything, see the example code I posted. The compound literal would have the same effect as your current code, except it's shorter, can be optimized by the compiler and will automatically initialize any fields added in the future to zero as well.